### ISSUE TITLE
 Callback not firing for incoming call when using GammuWorker

### ISSUE THREAD

**Issue by vicgordienko** [gammu/python-gammu#11]
When I use GammuWorker,  callback is not firing for incoming call events type. During incoming call it hangs.
For incoming SMS - ok.
Example bellow:

```
__author__ = 'vic'
#!/usr/bin/env python

import gammu
import time
import gammu.worker

def _event_callback(sm, type, data):
    print 'Received incoming event type %s, data:' % type
    print data

def _worker_callback(name, result, error, percents):
    print '-> %s completed %d%% with error %s , return value:' % (
            name,
            percents,
            error)
    print result
    pass

_gsm_config = {
        'StartInfo': True,
        'UseGlobalDebugFile': True,
        'DebugFile': None,
        'SyncTime': True,
        'Connection': 'at115200',
        'LockDevice': False,
        'DebugLevel': 'textalldate',
        'Device': '/dev/ttyUSB0',
        'Model': '',
        }

_sm = gammu.StateMachine()
_sm.SetConfig(0, _gsm_config)
_worker = gammu.worker.GammuWorker(_worker_callback)
_worker.configure(_gsm_config)
_worker.initiate()
_worker.enqueue('SetIncomingCallback', [_event_callback])
_worker.enqueue('SetIncomingCall')
_worker.enqueue('SetIncomingCB')
_worker.enqueue('SetIncomingSMS')


while(1):
    time.sleep(3)
    _worker.enqueue('ReadDevice')
    _worker.enqueue('GetSignalQuality')
```

 I will appreciate if you share any information.
Thank you.

**Comment by nijel** [comment]
What Gammu and python-gammu versions are you using? I think I've fixed recently something in this area, but I can't find exact fixes right now...

**Comment by vicgordienko** [comment]
print(sys.version)
2.7.6 (default, Jun 22 2015, 17:58:13) 
print gammu.Version()
('1.37.4', '2.6', '1.37.4')

**Comment by nijel** [comment]
Hmm, there is no relevant fix after these. Can you please provide debug log from the communication?

**Comment by vicgordienko** [comment]
I little bit modified code (moved SetIncomingXX after device has been initialized):
Log: [gsm.txt](https://github.com/gammu/python-gammu/files/534312/gsm.txt)

```
import gammu
import time
import gammu.worker

gsm_init = False

def _event_callback(sm, type, data):
    print 'Received incoming event type %s, data:' % type
    print data

def _worker_callback(name, result, error, percents):
    print '-> %s completed %d%% with error %s , return value:' % (
            name,
            percents,
            error)
    print result

    if name == 'Init' and error == 'ERR_NONE':
        _worker.enqueue('SetIncomingCallback', [_event_callback])
        _worker.enqueue('SetIncomingCall')
        _worker.enqueue('SetIncomingCB')
        _worker.enqueue('SetIncomingSMS')
        global gsm_init
        gsm_init = True

_gsm_config = {
        'StartInfo': True,
        'UseGlobalDebugFile': False,
        'DebugFile': '/home/vic/prj/app/log/gsm.txt',
        'SyncTime': True,
        'Connection': 'at115200',
        'LockDevice': False,
        'DebugLevel': 'textalldate',
        'Device': '/dev/ttyUSB0',
        'Model': '',
        }

_sm = gammu.StateMachine()
_sm.SetConfig(0, _gsm_config)
_worker = gammu.worker.GammuWorker(_worker_callback)
_worker.configure(_gsm_config)
_worker.initiate()

while(1):
    time.sleep(3)
    if gsm_init:
        _worker.enqueue('ReadDevice')
        _worker.enqueue('GetSignalQuality')
```

**Comment by nijel** [comment]
I've just added [test covering this scenario](https://github.com/gammu/python-gammu/blob/master/test/test_worker.py#L349) and the callback is invoked correctly.

Few remarks (though none of these should cause any problem, it just complicates the code):

* You don't have to invoke `ReadDevice` manually, it's done automatically when queue is empty.
* Commands can be enqueued at any time, so there is no need to do that in the callback.
* There is no need to construct StateMachine object when using worker.

However looking at the log there is one bug in Gammu library preventing correct handling of the incoming calls (and also causing other problems in the communication), I've fixed that in gammu/gammu@13618312fd47d39bfb6fa5bbe9cf13385fd2cb84.
